01-JS-DOM-APIs/fade-in.js

```diff
@@ -0,0 +1,115 @@
+const button = document.getElementById('button');
+let fadeInText = document.getElementById('fade-in-text');
+
+window.onload = function () {
+  fadeInText.setAttribute('class', 'visible-text');
+};
+
+let config = {
+  method: 'GET',
+  url: 'http://api.icndb.com/jokes/random'
+};
+
+button.addEventListener('click', htmlUpdate);
+
+function htmlUpdate() {
+  let ObjectOfData = promiseFunction(config);
+  ObjectOfData.then(function (result) {
+    result.send();
+    result.onload = function () {
+      let data = JSON.parse(result.responseText);
+      fadeInText.innerHTML = data.value.joke;
+      fadeInText.setAttribute('class', 'small-text');
+    }
+  }, function (err) {
+    let hiddenTextContainer = document.getElementById('hidden-text-container');
+    hiddenTextContainer.setAttribute('class', 'hidden-text-container-error');
+    fadeInText.setAttribute('class', 'small-text');
+    fadeInText.innerHTML = err;
+  });
+};
+
+function promiseFunction(configObject) {
+  let promise = new Promise(function (resolve, reject) {
+    let data = new XMLHttpRequest();
+    if ('withCredentials' in data) {
+      data.open(configObject.method, configObject.url, true);
+    } else if (typeof XDomainRequest !== 'undefined') {
+      data = new XDomainRequest();
+      data.open(configObject.method, configObject.url);
+    } else {
+      data = null;
+    }
+    if (data !== null) {
```

Comment:
make sense to use `if (!data) {...` instead? Have in mind null is falsy as undefined and 0, might be useful to know this here